src/F1_plotting.py

```diff
@@ -103,8 +104,11 @@ def extract_plot_data_and_title(dict_values, df_dem=None):
         df_dem = convert_demand_to_dataframe(dict_values)
 
     # Collect the keys of various resources (PV, Wind, etc.)
-    resources = dict_values[ENERGY_PRODUCTION]
-    res_keys = [k for k in resources.keys() if "DSO_" not in k]
+    resources = dict_values[ENERGY_PRODUCTION].copy()
+    res_keys = []
+    for k in resources.keys():
+        if resources[k][DISPATCHABILITY] == False:
```

Comment:
```suggestion
        if resources[k][DISPATCHABILITY] is False:
```
The one line formulation is not to your taste?
`res_keys = [k for k in resources.keys() if resources[k][DISPATCHABILITY] is False]`

Reply:
Hm, actually I just can never come up with these formulations, they are too complex for me. Fixed in 26fe3f864bbe908876a73057b6e3dac5dd040b47, PR #525